### DISCUSSION TITLE
MongoRemotePeerHelper.getRemotePeer may block the process

### DISCUSSION THREAD

**Discussion by shichaoyuan** [apache/skywalking#11201] [General]
In MongoRemotePeerHelper.getRemotePeer, `cluster.getDescription()` triggers a remote call, it may block for MaxWaitTime.

Can it be replaced with `SrvHost` and `Hosts` of `Settings` ？

just a native call

**Comment by wu-sheng** [comment]
What are you saying? You need to be more specific rather than so general.

**Reply by shichaoyuan** [reply]
https://github.com/apache/skywalking-java/blob/e7394a87da5d2ec4e519327c8296b24ab02255d1/apm-sniffer/apm-sdk-plugin/mongodb-4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v4/support/MongoRemotePeerHelper.java#L36-L43

![image](https://github.com/apache/skywalking/assets/1159482/defc2ec2-068b-4a15-ac51-0729fad7e8d8)

 remote call in `cluster.getDescription()`

**Reply by wu-sheng** [reply]
> Can it be replaced with SrvHost and Hosts of Settings?

So, by replacing, what are the differences? The `MongoRemotePeerHelper#getRemotePeer` is only called on the constructor interceptor, which should not affect much time.
What is the issue? When the client is built, the server is not connectable?

**Reply by shichaoyuan** [reply]
Mainly we have a lot of legacy code, some mongo are no longer available.

If this code is injected, it will lead to increased startup time.

**Reply by wu-sheng** [reply]
Then, what is the point of upstream changing? We are going to change if something is wrong, you can't ask the community to pay for your own/private tech debt. That is totally unfair.
